src/session_impl.cpp

```diff
@@ -5570,9 +5581,6 @@ namespace {
 			return std::uint16_t(sock->tcp_external_port());
 		}
 
-		if (m_settings.get_int(settings_pack::proxy_type) != settings_pack::none
-			&& m_settings.get_bool(settings_pack::proxy_peer_connections))
-			return 0;
```

Comment:
This doesn't look right. This should probably mimic the `session_impl::listen_port()` above. I think you'd want to update the check and early exit on line 5569 as well